README.md

```diff
@@ -41,15 +41,35 @@ const { stop } = blocked(fn, options)
 |---|---|---|
 |`trimFalsePositives`|*falsy*| eliminate a class of false positives (experimental) |
 |`threshold`| *20* | minimum miliseconds of blockage to report. supported for parity with [`blocked`](https://www.npmjs.com/package/blocked)|
+|`maxResourcesSize`| *undefined* | maximum amount of stack traces with resource details kept in memory. see the next section for details |
 |`debug`| *falsy* | print debug data to console |
 
 Returns: An object with `stop` method. `stop()` will disable the async hooks set up by this library and callback will no longer be called.
 
-## Using the stack trace
+## Using the stack trace and resource details
 
 The stack trace is pointing to a start of a function called asynchronously, so in most cases the first stack frame pointing to your code is where you need to start analyzing all synchronous operations to find the slow one.
 
-In some cases your code is not directly called and tracking it down will still be difficult. See how the http test case produces a stack pointing to `Server.connectionListener` as the slow function, because everything inside of it is synchronously called. You can always wrap your handlers' code in `setImmediate` if you become desperate.
+In some cases your code is not directly called and tracking it down will still be difficult. See how the http test case produces a stack pointing to `Server.connectionListener` as the slow function, because everything inside of it is synchronously called. You can try to narrow down your search by using `maxResourcesSize` option and inspecting an associated [resource](https://nodejs.org/api/async_hooks.html#async_hooks_resource):
+
+ ```js
+blocked((time, stack, {type, resource}) => {
+  console.log(`Blocked for ${time}ms, operation started here:`, stack)
+  if (type === 'HTTPPARSER' && resource) {
+    // resource structure assumes Node 10.x
+    console.log(`URL related to blocking operation: ${resource.resource.incoming.url}`)
+  }
+}, {maxResourcesSize: 100})
+```
+
+Note that resource structure is a subject to change and may vary between Node versions.
+
+ After you've identified a problematic URL, you can wrap your handlers' code in `setImmediate` which should make the stack point to something meaningful.
```

Comment:
I'd like to keep the set* suggestion as the default thing to try. It's more likely to work if you already have a suspect.
I'd move it back up and describe resources usecase as a separate paragraph.